### PR TITLE
Fixes #13162 : added autofocus on plugin marketplace modal open

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
When the admin open the plugin marketplace modal, The main search input now autofocus, so user starts typing rightaway

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13162
JIRA ticket MM-20304

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@admin should test for focus be included

File: components/plugin_marketplace/marketplace_modal.js [hunk 1]
```diff
@@ -141,6 +142,11 @@ export class MarketplaceModal extends React.Component {
         trackEvent('plugins', 'ui_marketplace_opened');
 
         this.fetchPlugins();
+
+        // Do not autofocus on mobile
+        if (!UserAgent.isMobile() && this.refs.filter) {
```
[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply components/plugin_marketplace/marketplace_modal.js]: Got it will remove it

File: components/plugin_marketplace/marketplace_modal.js [hunk 2]
```diff
@@ -141,6 +141,11 @@ export class MarketplaceModal extends React.Component {
         trackEvent('plugins', 'ui_marketplace_opened');
 
         this.fetchPlugins();
+
+        // Do not autofocus on mobile
```
Comment (marianunez) [inline components/plugin_marketplace/marketplace_modal.js]: I believe this comment is no longer valid

**Comment by ali-farooq0** [comment]
@saturninoabril this is ready for QA.

**Pull request merged**